t/t2291-job-update-queue.t

```diff
@@ -0,0 +1,152 @@
+#!/bin/sh
+test_description='Test update of job queue'
+
+. $(dirname $0)/sharness.sh
+
+if flux job submit --help 2>&1 | grep -q sign-type; then
+	test_set_prereq HAVE_FLUX_SECURITY
+fi
+
+test_under_flux 4 full
+
+flux setattr log-stderr-level 1
+
+runas_guest() {
+	local userid=$(($(id -u)+1))
+        FLUX_HANDLE_USERID=$userid FLUX_HANDLE_ROLEMASK=0x2 "$@"
+}
+
+submit_held_job_as_guest()
+{
+	local duration=$1
+	local userid=$(($(id -u)+1))
+        flux run --dry-run -t $duration true | \
+          flux python ${SHARNESS_TEST_SRCDIR}/scripts/sign-as.py $userid \
+            >job.signed
+        FLUX_HANDLE_USERID=$userid \
+          flux job submit --flags=signed --urgency=0 job.signed
+}
+
+test_expect_success 'config queues and resources' '
+	flux R encode -r 0-3 -p batch:0-3 -p debug:3 \
+	   | tr -d "\n" \
+	   | flux kvs put -r resource.R=- &&
+	flux config load <<-EOT &&
+	[queues.batch]
+	requires = [ "batch" ]
+	policy.limits.duration = "1h"
+	policy.jobspec.defaults.system.duration = "1h"
+
+	[queues.debug]
+	requires = [ "debug" ]
+	policy.limits.duration = "1m"
+	policy.jobspec.defaults.system.duration = "1m"
+
+	[policy.jobspec.defaults.system]
+	queue = "batch"
+	EOT
+	flux queue start --all &&
+	flux module unload sched-simple &&
+	flux module reload resource &&
+	flux module load sched-simple &&
+	flux queue list &&
+	flux resource list -o rlist
+'
```

Comment:
should we add a variant test where there are no requires/constraints within a queue?

Reply:
Oh, yeah. I swear I did that, I wonder if I just didn't push the result. Let me look around...

Reply:
Oops, looks like it got squashed into the wrong commit :facepalm: I'll fix that.

Reply:
Ok, moved the changes to the right commit and force-pushed the result.

Reply:
changes LGTM!